src/main/java/ch/njol/skript/effects/EffToggle.java

```diff
@@ -40,63 +39,88 @@
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
 
-/**
- * @author Peter Güttinger
- */
-@SuppressWarnings("deprecation")
 @Name("Toggle")
-@Description("Toggle the state of a block.")
+@Description("Toggle a boolean or the state of a block.")
 @Examples({"# use arrows to toggle switches, doors, etc.",
 		"on projectile hit:",
 		"\tprojectile is arrow",
-		"\ttoggle the block at the arrow"})
-@Since("1.4")
+		"\ttoggle the block at the arrow",
+		"",
+		"# With booleans",
+		"toggle gravity of player"
+})
+@Since("1.4, INSERT VERSION (booleans)")
 public class EffToggle extends Effect {
 	
 	static {
-		Skript.registerEffect(EffToggle.class, "(close|turn off|de[-]activate) %blocks%", "(toggle|switch) [[the] state of] %blocks%", "(open|turn on|activate) %blocks%");
+		Skript.registerEffect(EffToggle.class,
+			"(open|turn on|activate) %blocks%",
+			"(close|turn off|de[-]activate) %blocks%",
+			"(toggle|switch) [[the] state of] %booleans/blocks%");
+	}
+
+	private enum State {
+		ACTIVATE, DEACTIVATE, TOGGLE
 	}
 
 	@SuppressWarnings("null")
-	private Expression<Block> blocks;
-	private int toggle;
+	private Expression<?> togglables;
+	private State state;
 	
-	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] vars, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		blocks = (Expression<Block>) vars[0];
-		toggle = matchedPattern - 1;
+	public boolean init(Expression<?>[] vars, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		togglables = (Expression<?>) vars[0];
+		state = State.values()[matchedPattern];
+		if (togglables.getReturnType() == Boolean.class && !ChangerUtils.acceptsChange(togglables, ChangeMode.SET, Boolean.class)) {
```

Comment:
We may actually need to check that it accepts Boolean[].class when `toggleables.isSingle()` is false

Reply:
Do we really need to have this kind of check? Doesn't the condition already manage it?

Reply:
Well, there are multiple cases we probably want to check here:

1. the expression does not accept booleans (this is already done)
2. the expression returns multiple booleans but can only be set to one

If the expression returns multiple booleans and can only be set to one, we wouldn't know which to actually set it to, and thus need to error here. You can check like:
```java
(!togglables.isSingle() && !ChangerUtils.acceptsChange(togglables, ChangeMode.SET, Boolean[].class))
```